cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/VersionInfo.java

```diff
@@ -20,7 +20,66 @@
  * Version information that is comparable to other instance of the same class or any other Object that
  * returns properly formatted version from it's {@link Object#toString()}. E.g. you can compare it to a string.
  * Also any instance of this interface must return proper bare version from it's toString.
+ * Default implementation assumes version of the form "major.minor.fix-buildTime"/"major.minor.fix-SNAPSHOT-buildTime".
  */
 public interface VersionInfo extends Comparable<Object> {
 
+  default int getMajor() {
+    String[] versionInfo = this.toString().split("-");
```

Comment:
Better to have a comment about the default implementation of why using string split.

Reply:
added the comment in javadoc.